src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```diff
@@ -0,0 +1,146 @@
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.ec2;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.ec2.AmazonEC2Client;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusResult;
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.InstanceState;
+import com.amazonaws.services.ec2.model.RunInstancesRequest;
+import com.jcabi.aspects.Immutable;
+import com.jcabi.aspects.Tv;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Amazon EC2 instance on demand.
+ *
+ * @author Marton Horvath (marton.horvath@m323.org)
+ * @version $Id$
+ * @since 2.0
+ */
+@Immutable
+public class AmazonEC2 implements Amazon {
+    /**
+     * The integer value of the Running {@link InstanceState}.
+     */
+    private static final int RUNNING_STATE = 16;
+    /**
+     * AWS credentials.
+     */
+    private final transient AWSCredentials credentials;
+    /**
+     * AWS region.
+     */
+    private final transient String region;
+    /**
+     * EC2 instance type.
+     */
+    private final transient String type;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials
+     * @param zone AWS region
+     * @param instype Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String instype,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.type = instype;
+        this.keyname = key;
+    }
+
+    @SuppressWarnings("PMD.ConfusingTernary")
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance;
+        final AmazonEC2Client client = new AmazonEC2Client(
+            this.credentials
+        );
+        client.setEndpoint(this.region);
+        final List<Instance> instances = client.runInstances(
+            this.create()
+        ).getReservation().getInstances();
+        if ((instances != null) && !instances.isEmpty()) {
```

Comment:
also, please also invert the if check ( `not` checks are confusing, that's why PMD complains) and remove Suppress from above
So the if would look like this:
```
if (instances.isEmpty()) {
    throw new IllegalStateException("No Instance was available in the RunInstanceRequest reply!");
} else {
    instance = instances.get(0);
}
```